cpp/src/io/parquet/reader_impl_preprocess.cu

```diff
@@ -280,6 +280,70 @@ void generate_depth_remappings(std::map<int, std::pair<std::vector<int>, std::ve
   return total_pages;
 }
 
+/**
+ * @brief Returns a string representation of known encodings
+ *
+ * @param encoding Given encoding
+ * @return STring representation of encoding
+ */
+std::string encoding_to_string(Encoding encoding)
+{
+  switch (encoding) {
+    case Encoding::PLAIN: return "PLAIN";
+    case Encoding::GROUP_VAR_INT: return "GROUP_VAR_INT";
+    case Encoding::PLAIN_DICTIONARY: return "PLAIN_DICTIONARY";
+    case Encoding::RLE: return "RLE";
+    case Encoding::BIT_PACKED: return "BIT_PACKED";
+    case Encoding::DELTA_BINARY_PACKED: return "DELTA_BINARY_PACKED";
+    case Encoding::DELTA_LENGTH_BYTE_ARRAY: return "DELTA_LENGTH_BYTE_ARRAY";
+    case Encoding::DELTA_BYTE_ARRAY: return "DELTA_BYTE_ARRAY";
+    case Encoding::RLE_DICTIONARY: return "RLE_DICTIONARY";
+    case Encoding::BYTE_STREAM_SPLIT: return "BYTE_STREAM_SPLIT";
+    case Encoding::NUM_ENCODINGS: return "NUM_ENCODINGS";
+    default: return "UNKNOWN";
+  }
+}
+
+/**
+ * @brief Helper function to convert an encoding bitmask to a readable string
+ *
+ * @param bitmask Bitmask of found unsupported encodings
+ * @returns Human readable string with unsupported encodings
+ */
+std::string encoding_bitmask_to_str(int32_t encoding_bitmask)
+{
+  std::string result;
+  for (int32_t i = 0; i < 32; ++i) {
```

Comment:
Lost a comment :( I was wondering if this could use `bitset<uint32_t>`. Then the loop could use `bitset.size()` as the upper bound, and you could use `bitset.test(i)` to check bit positions. Removes one magic number :)

Reply:
Sounds good! I'm a scala programmer by day so I might miss some things as I rarely get to program at a low level 😅. Will change now.